ntoskrnl/ke/amd64/traphandler.c

```diff
@@ -139,11 +139,11 @@ KiSystemCallHandler(
 
     /* Get the system call number from the trap frame and decode it */
     ServiceNumber = (ULONG)TrapFrame->Rax;
-    Offset = (ServiceNumber >> SERVICE_TABLE_SHIFT) & SERVICE_TABLE_MASK;
+    TableIndex = (ServiceNumber >> TABLE_OFFSET_BITS) & ((1 << TABLE_NUMBER_BITS) - 1);
```

Comment:
I'm not fully convinced of the necessity of this change (and the departure from what is done in the other architectures).